tests/test_data.py

```diff
@@ -148,3 +148,20 @@ def test_exercise_to_dict(exercise, exercise_data):
         assert key in exercise_dict, "Key {} is not found in to_dict method".format(key)
     for key, value in exercise_dict.items():
         assert value == exercise_data.get(key), "Mismatched {}: {} != {}".format(key, value, exercise_data[key])
+
+def test_slideshow_to_dict(slideshow, slideshow_data):
+    slideshow_dict = slideshow.to_dict()
+    extra_fields = json.loads(slideshow_dict['extra_fields'])
+    assert len(extra_fields['slideshow_data']) == 10, 'wrong num slides'
+    # TODO check extra_fields
```

Comment:
It might be useful to add some checks here to make sure the `extra_fields['slideshow_data']` dict has appropriate fields --- explicitly check the data we send is what Studio expects.

Reply:
@ivanistheone More on the ineffectiveness of the `assert(filter...` 

I tried checking all of the slideshow_data items like so (written in place of your TODO highlighted above):

```
expected_field_keys = { 'caption', 'descriptive_text', 'checksum', 'sort_order', 'extension'}
assert all(filter(lambda f: set(f.keys()) == expected_field_keys), extra_fields['slideshow_data'])), "error message"
```

But when I set the second line to NOT EQUAL like:

```
expected_field_keys = { 'caption', 'descriptive_text', 'checksum', 'sort_order', 'extension'}
assert all(filter(lambda f: set(f.keys()) != expected_field_keys), extra_fields['slideshow_data'])), "error message"
```

The tests still passed.

However, this:

```
expected_field_keys = { 'caption', 'descriptive_text', 'checksum', 'sort_order', 'extension'}
    for slide_data in extra_fields['slideshow_data']:
        assert set(slide_data.keys()) == expected_field_keys, 'extra_fields is missing expected fields'
```

Fails as expected when I switch the logic.

Am I not understanding the `assert all(filter())` code somehow?

Reply:
> Am I not understanding the assert all(filter()) code somehow?

Well it's not so much a misunderstanding as it's not necessary --- I think I may have copy-pasted the code from another place filter was useful but here you don't need.

The check here check could be
```
assert all([set(sd.keys()) == expected_field_keys for sd in extra_fields['slideshow_data']]), 'extra_fields is missing expected fields'
```
which is exactly the same as the for loop:
```
    for slide_data in extra_fields['slideshow_data']:
        assert set(slide_data.keys()) == expected_field_keys, 'extra_fields is missing expected fields'
```